src/cmd/services/m3comparator/main/series_load_handler.go

```diff
@@ -0,0 +1,201 @@
+}
+
+var _ http.Handler = (*seriesLoadHandler)(nil)
+
+// newSeriesLoadHandler builds a handler that can load series
+// to the comparator via an http endpoint.
+func newSeriesLoadHandler(iterOpts iteratorOptions) *seriesLoadHandler {
+	return &seriesLoadHandler{
+		iterOpts:        iterOpts,
+		nameIDSeriesMap: make(nameIDSeriesMap),
+	}
+}
+
+func (l *seriesLoadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	logger := l.iterOpts.iOpts.Logger()
+	err := l.serveHTTP(r)
+	if err != nil {
+		logger.Error("unable to fetch data", zap.Error(err))
+		xhttp.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (l *seriesLoadHandler) getSeriesIterators(
+	name string,
+) encoding.SeriesIterators {
+	l.RLock()
+	defer l.RUnlock()
+
+	logger := l.iterOpts.iOpts.Logger()
+	seriesMap, found := l.nameIDSeriesMap[name]
+	if !found || len(seriesMap.series) == 0 {
+		return nil
+	}
+
+	iters := make([]encoding.SeriesIterator, 0, len(seriesMap.series))
+	for _, series := range seriesMap.series {
+		encoder := l.iterOpts.encoderPool.Get()
+		dps := series.Datapoints
+		encoder.Reset(time.Time{}, len(dps), nil)
+		for _, dp := range dps {
+
+			err := encoder.Encode(ts.Datapoint{
+				Timestamp:      dp.Timestamp,
+				Value:          dp.Value,
+				TimestampNanos: xtime.ToUnixNano(dp.Timestamp),
+			}, xtime.Second, nil)
+
+			if err != nil {
+				encoder.Close()
+				logger.Error("error encoding datapoints", zap.Error(err))
+				return nil
+			}
+		}
+
+		readers := [][]xio.BlockReader{{{
+			SegmentReader: xio.NewSegmentReader(encoder.Discard()),
+			Start:         series.Start,
+			BlockSize:     series.End.Sub(series.Start),
+		}}}
+
+		multiReader := encoding.NewMultiReaderIterator(
+			iterAlloc,
+			l.iterOpts.iteratorPools.MultiReaderIterator(),
+		)
+
+		sliceOfSlicesIter := xio.NewReaderSliceOfSlicesFromBlockReadersIterator(readers)
+		multiReader.ResetSliceOfSlices(sliceOfSlicesIter, nil)
+		tagIter, id := buildTagIteratorAndID(tagMap(series.Tags), l.iterOpts.tagOptions)
+		iters = append(iters, encoding.NewSeriesIterator(
+			encoding.SeriesIteratorOptions{
+				ID:             id,
+				Namespace:      ident.StringID("ns"),
+				Tags:           tagIter,
+				StartInclusive: xtime.ToUnixNano(series.Start),
+				EndExclusive:   xtime.ToUnixNano(series.End),
+				Replicas: []encoding.MultiReaderIterator{
+					multiReader,
+				},
+			}, nil))
+	}
+
+	return encoding.NewSeriesIterators(
+		iters,
+		l.iterOpts.iteratorPools.MutableSeriesIterators(),
+	)
+}
+
+func (l *seriesLoadHandler) serveHTTP(r *http.Request) error {
+	l.Lock()
+	defer l.Unlock()
+
+	logger := l.iterOpts.iOpts.Logger()
+	if err := r.ParseForm(); err != nil {
+		return err
+	}
+
+	series := r.Form["series"]
+	if len(series) != 1 {
+		logger.Info("wrong length in series")
+		return nil
+	}
+
+	buf := []byte(series[0])
+	seriesList := make([]parser.Series, 0, 10)
+	if err := json.Unmarshal(buf, &series); err != nil {
```

Comment:
Hm, any reason not to just unmarshal the whole request body rather than something in the "series" encoded form contents? 

i.e. 
```go
type seriesLoadRequest struct {
  Series []parser.Series `json:"series"`
}

func (l *seriesLoadHandler) serveHTTP(r *http.Request) error {
  defer r.Body.Close()
 
  var req seriesLoadRequest
  if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
    // ....
  } 
```

Just seems slighlty non-standard to encode to JSON, then to encode to multipart-form, rather than just single level of serialization (i.e. just JSON).